sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Iot.ModelsRepository.Fetchers;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Iot.ModelsRepository
+{
+    internal class RepositoryHandler
+    {
+        private readonly IModelFetcher _modelFetcher;
+        private readonly Guid _clientId;
+
+        public Uri RepositoryUri { get; }
+        public ResolverClientOptions ClientOptions { get; }
+
+        public RepositoryHandler(Uri repositoryUri, ResolverClientOptions options = null)
+        {
+            ClientOptions = options ?? new ResolverClientOptions();
+            RepositoryUri = repositoryUri;
+            _modelFetcher = repositoryUri.Scheme == "file" ?
+                _modelFetcher = new LocalModelFetcher(ClientOptions) :
+                _modelFetcher = new RemoteModelFetcher(ClientOptions);
+            _clientId = Guid.NewGuid();
+            ResolverEventSource.Shared.InitFetcher(_clientId, repositoryUri.Scheme);
+        }
+
+        public async Task<IDictionary<string, string>> ProcessAsync(string dtmi, CancellationToken cancellationToken)
+        {
+            return await ProcessAsync(new List<string>() { dtmi }, cancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task<IDictionary<string, string>> ProcessAsync(IEnumerable<string> dtmis, CancellationToken cancellationToken)
+        {
+            Dictionary<string, string> processedModels = new Dictionary<string, string>();
+            Queue<string> toProcessModels = new Queue<string>();
+
+            foreach (string dtmi in dtmis)
+            {
+                if (!DtmiConventions.IsDtmi(dtmi))
+                {
+                    ResolverEventSource.Shared.InvalidDtmiInput(dtmi);
+                    string invalidArgMsg = string.Format(CultureInfo.InvariantCulture, StandardStrings.InvalidDtmiFormat, dtmi);
```

Comment:
If we're putting strings in a resx file, why not use CurrentCulture?

Reply:
That is an outstanding conversation I need to have with Paymaun about the Culture info. There is a topic on my onenote to go over with him.